packages/caravan-health/src/privacy.ts

```diff
@@ -0,0 +1,261 @@
+import { SpendType, MultisigAddressType, Network, Transaction } from "./types";
+import { getAddressType } from "@caravan/bitcoin";
+import { WalletMetrics } from "./wallet";
+import { determineSpendType, getSpendTypeScore } from "./utility";
+
+// Deniability Factor is a normalizing quantity that increases the score by a certain factor in cases of self-payment.
+// More about deniability : https://www.truthcoin.info/blog/deniability/
+const DENIABILITY_FACTOR = 1.5;
+
+export class PrivacyMetrics extends WalletMetrics {
+  /*
+    Name : Topology Score
+
+    Definition :
+      The score is calculated based on the number of inputs and outputs which 
+      influence the topology type of the transaction.
+
+    Calculation :
+      We have 5 categories of transaction type each with their own impact on privacy score
+      - Perfect Spend (1 input, 1 output)
+      - Simple Spend (1 input, 2 outputs)
+      - UTXO Fragmentation (1 input, more than 2 standard outputs)
+      - Consolidation (more than 1 input, 1 output)
+      - CoinJoin or Mixing (more than 1 input, more than 1 output)
+  */
+  getTopologyScore(transaction: Transaction): number {
```

Comment:
This could also be used in the preview transaction view right?

Reply:
Yes, We can give some degree that how it will behave privacy wise if this previewed transaction is actually made.